sql: TestBackfillWithProtectedTS failed

sql.TestBackfillWithProtectedTS [failed](https://tanzanite.cluster.engflow.com/invocations/default/f84dcd64-01eb-4263-996c-7b76ec38f25e?testReportRun=6&testReportShard=14&testReportAttempt=1#targets-Ly9wa2cvc3FsOnNxbF90ZXN0) on master @ [12a979cd5f2ac1d37502a1353596a1d77751ab5d](https://github.com/cockroachdb/cockroach/commits/12a979cd5f2ac1d37502a1353596a1d77751ab5d):


```
=== RUN   TestBackfillWithProtectedTS
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestBackfillWithProtectedTS3546642436
    test_log_scope.go:76: use -show-logs to present logs inline
    test_server_shim.go:141: cluster virtualization disabled in global scope due to issue: #76378 (expected label: C-bug)
    backfill_protected_timestamp_test.go:526: -- test log scope end --
test logs left over in: outputs.zip/logTestBackfillWithProtectedTS3546642436
--- FAIL: TestBackfillWithProtectedTS (500.31s)
=== RUN   TestBackfillWithProtectedTS/create_index
    backfill_protected_timestamp_test.go:477: running backfill with PTS not setup early enough
    backfill_protected_timestamp_test.go:425: i=0 begin running operations in the middle of backfill
    backfill_protected_timestamp_test.go:429: i=0 finished running operations in the middle of backfill. retErr: failed to DELETE and INSERT: pq: batch timestamp 1737812360.241467438,0 must be after replica GC threshold 1737812367.650912440,0 (r78: /Table/107{-/2})
    backfill_protected_timestamp_test.go:474: backfill goroutine failed: could not read jobs table: context canceled
    backfill_protected_timestamp_test.go:515: 
        	Error Trace:	pkg/sql/backfill_protected_timestamp_test.go:515
        	Error:      	Received unexpected error:
        	            	failed to DELETE and INSERT: pq: batch timestamp 1737812360.241467438,0 must be after replica GC threshold 1737812367.650912440,0 (r78: /Table/107{-/2})
        	            	(1) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/sql_test.TestBackfillWithProtectedTS.func7.2.1
        	            	  | 	pkg/sql/backfill_protected_timestamp_test.go:438
        	            	  | github.com/cockroachdb/cockroach/pkg/sql_test.TestBackfillWithProtectedTS.func7.2
        	            	  | 	pkg/sql/backfill_protected_timestamp_test.go:463
        	            	  | github.com/cockroachdb/cockroach/pkg/sql_test.TestBackfillWithProtectedTS.func7.Group.GoCtx.4
        	            	  | 	pkg/util/ctxgroup/ctxgroup.go:189
        	            	  | golang.org/x/sync/errgroup.(*Group).Go.func1
        	            	  | 	external/org_golang_x_sync/errgroup/errgroup.go:78
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_amd64.s:1695
        	            	Wraps: (2) failed to DELETE and INSERT
        	            	Wraps: (3) pq: batch timestamp 1737812360.241467438,0 must be after replica GC threshold 1737812367.650912440,0 (r78: /Table/107{-/2})
        	            	Error types: (1) *withstack.withStack (2) *errutil.withPrefix (3) *pq.Error
        	Test:       	TestBackfillWithProtectedTS/create_index
    --- FAIL: TestBackfillWithProtectedTS/create_index (439.84s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=6</code>
 - <code>shard=14</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestBackfillWithProtectedTS.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-46832

sql.TestBackfillWithProtectedTS [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/18641914?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/18641914?buildTab=artifacts#/) on master @ [aba435dba18db3094a191166212c951e25b5be35](https://github.com/cockroachdb/cockroach/commits/aba435dba18db3094a191166212c951e25b5be35):


```
=== RUN   TestBackfillWithProtectedTS
    test_log_scope.go:165: test logs captured to: /artifacts/tmp/_tmp/ebab9da743cf8d3e9fbccfacc75c4814/logTestBackfillWithProtectedTS1516674431
    test_log_scope.go:76: use -show-logs to present logs inline
    test_server_shim.go:141: cluster virtualization disabled in global scope due to issue: #76378 (expected label: C-bug)
    backfill_protected_timestamp_test.go:526: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/ebab9da743cf8d3e9fbccfacc75c4814/logTestBackfillWithProtectedTS1516674431
--- FAIL: TestBackfillWithProtectedTS (473.07s)
=== RUN   TestBackfillWithProtectedTS/create_index
    backfill_protected_timestamp_test.go:477: running backfill with PTS not setup early enough
    backfill_protected_timestamp_test.go:425: i=0 begin running operations in the middle of backfill
    backfill_protected_timestamp_test.go:429: i=0 finished running operations in the middle of backfill. retErr: failed to DELETE and INSERT: pq: batch timestamp 1738100967.884760635,0 must be after replica GC threshold 1738101006.109462831,0 (r79: /{Table/107/3-Max})
    backfill_protected_timestamp_test.go:474: backfill goroutine failed: expected error was not hit
    backfill_protected_timestamp_test.go:515: 
        	Error Trace:	pkg/sql/backfill_protected_timestamp_test.go:515
        	Error:      	Received unexpected error:
        	            	failed to DELETE and INSERT: pq: batch timestamp 1738100967.884760635,0 must be after replica GC threshold 1738101006.109462831,0 (r79: /{Table/107/3-Max})
        	            	(1) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/sql_test.TestBackfillWithProtectedTS.func7.2.1
        	            	  | 	pkg/sql/backfill_protected_timestamp_test.go:438
        	            	  | github.com/cockroachdb/cockroach/pkg/sql_test.TestBackfillWithProtectedTS.func7.2
        	            	  | 	pkg/sql/backfill_protected_timestamp_test.go:463
        	            	  | github.com/cockroachdb/cockroach/pkg/sql_test.TestBackfillWithProtectedTS.func7.Group.GoCtx.4
        	            	  | 	pkg/util/ctxgroup/ctxgroup.go:189
        	            	  | golang.org/x/sync/errgroup.(*Group).Go.func1
        	            	  | 	external/org_golang_x_sync/errgroup/errgroup.go:78
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_arm64.s:1222
        	            	Wraps: (2) failed to DELETE and INSERT
        	            	Wraps: (3) pq: batch timestamp 1738100967.884760635,0 must be after replica GC threshold 1738101006.109462831,0 (r79: /{Table/107/3-Max})
        	            	Error types: (1) *withstack.withStack (2) *errutil.withPrefix (3) *pq.Error
        	Test:       	TestBackfillWithProtectedTS/create_index
    --- FAIL: TestBackfillWithProtectedTS/create_index (442.19s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #139991 sql: TestBackfillWithProtectedTS failed [C-test-failure O-robot T-sql-foundations branch-release-24.2 release-blocker]
- #139901 sql: TestBackfillWithProtectedTS failed [C-test-failure O-robot P-2 T-sql-foundations branch-release-24.3]
- #139862 sql: TestBackfillWithProtectedTS failed [C-test-failure O-robot P-1 T-sql-foundations branch-release-25.1 release-blocker]
- #139829 sql: TestBackfillWithProtectedTS failed [C-test-failure O-robot P-3 T-sql-foundations branch-release-24.3.4-rc]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestBackfillWithProtectedTS.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

sql.TestBackfillWithProtectedTS [failed](https://tanzanite.cluster.engflow.com/invocations/default/0a010f38-6694-4cc5-bf1a-c68d48b61d14?testReportRun=23&testReportShard=14&testReportAttempt=1#targets-Ly9wa2cvc3FsOnNxbF90ZXN0) on master @ [f6a6226821bbf5999381baa8659b6d7d93b3a526](https://github.com/cockroachdb/cockroach/commits/f6a6226821bbf5999381baa8659b6d7d93b3a526):


```
=== RUN   TestBackfillWithProtectedTS
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestBackfillWithProtectedTS3656757062
    test_log_scope.go:76: use -show-logs to present logs inline
    test_server_shim.go:141: cluster virtualization disabled in global scope due to issue: #76378 (expected label: C-bug)
    backfill_protected_timestamp_test.go:526: -- test log scope end --
test logs left over in: outputs.zip/logTestBackfillWithProtectedTS3656757062
--- FAIL: TestBackfillWithProtectedTS (498.17s)
=== RUN   TestBackfillWithProtectedTS/create_index
    backfill_protected_timestamp_test.go:477: running backfill with PTS not setup early enough
    backfill_protected_timestamp_test.go:425: i=0 begin running operations in the middle of backfill
    backfill_protected_timestamp_test.go:429: i=0 finished running operations in the middle of backfill. retErr: failed to DELETE and INSERT: pq: batch timestamp 1738133331.204352610,0 must be after replica GC threshold 1738133343.817533903,0 (r78: /Table/107{-/2})
    backfill_protected_timestamp_test.go:474: backfill goroutine failed: could not read jobs table: context canceled
    backfill_protected_timestamp_test.go:515: 
        	Error Trace:	pkg/sql/backfill_protected_timestamp_test.go:515
        	Error:      	Received unexpected error:
        	            	failed to DELETE and INSERT: pq: batch timestamp 1738133331.204352610,0 must be after replica GC threshold 1738133343.817533903,0 (r78: /Table/107{-/2})
        	            	(1) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/sql_test.TestBackfillWithProtectedTS.func7.2.1
        	            	  | 	pkg/sql/backfill_protected_timestamp_test.go:438
        	            	  | github.com/cockroachdb/cockroach/pkg/sql_test.TestBackfillWithProtectedTS.func7.2
        	            	  | 	pkg/sql/backfill_protected_timestamp_test.go:463
        	            	  | github.com/cockroachdb/cockroach/pkg/sql_test.TestBackfillWithProtectedTS.func7.Group.GoCtx.4
        	            	  | 	pkg/util/ctxgroup/ctxgroup.go:189
        	            	  | golang.org/x/sync/errgroup.(*Group).Go.func1
        	            	  | 	external/org_golang_x_sync/errgroup/errgroup.go:78
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_amd64.s:1695
        	            	Wraps: (2) failed to DELETE and INSERT
        	            	Wraps: (3) pq: batch timestamp 1738133331.204352610,0 must be after replica GC threshold 1738133343.817533903,0 (r78: /Table/107{-/2})
        	            	Error types: (1) *withstack.withStack (2) *errutil.withPrefix (3) *pq.Error
        	Test:       	TestBackfillWithProtectedTS/create_index
    --- FAIL: TestBackfillWithProtectedTS/create_index (442.84s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=23</code>
 - <code>shard=14</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #140043 sql: TestBackfillWithProtectedTS failed [C-test-failure O-robot T-sql-foundations branch-release-24.1 release-blocker]
- #139991 sql: TestBackfillWithProtectedTS failed [C-test-failure O-robot P-2 T-sql-foundations branch-release-24.2]
- #139901 sql: TestBackfillWithProtectedTS failed [C-test-failure O-robot P-2 T-sql-foundations branch-release-24.3]
- #139862 sql: TestBackfillWithProtectedTS failed [C-test-failure GA-blocker O-robot P-1 T-sql-foundations branch-release-25.1]
- #139829 sql: TestBackfillWithProtectedTS failed [C-test-failure O-robot P-3 T-sql-foundations branch-release-24.3.4-rc]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestBackfillWithProtectedTS.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

sql.TestBackfillWithProtectedTS [failed](https://tanzanite.cluster.engflow.com/invocations/default/c626743f-d224-487c-b56f-5ace71d839f4?testReportRun=25&testReportShard=14&testReportAttempt=1#targets-Ly9wa2cvc3FsOnNxbF90ZXN0) on master @ [8655f5db5fe8ff856d90f71211a3158f60f1769f](https://github.com/cockroachdb/cockroach/commits/8655f5db5fe8ff856d90f71211a3158f60f1769f):


```
=== RUN   TestBackfillWithProtectedTS
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestBackfillWithProtectedTS1110555499
    test_log_scope.go:76: use -show-logs to present logs inline
    test_server_shim.go:141: cluster virtualization disabled in global scope due to issue: #76378 (expected label: C-bug)
    backfill_protected_timestamp_test.go:526: -- test log scope end --
test logs left over in: outputs.zip/logTestBackfillWithProtectedTS1110555499
--- FAIL: TestBackfillWithProtectedTS (488.49s)
=== RUN   TestBackfillWithProtectedTS/create_index
    backfill_protected_timestamp_test.go:477: running backfill with PTS not setup early enough
    backfill_protected_timestamp_test.go:425: i=0 begin running operations in the middle of backfill
    backfill_protected_timestamp_test.go:429: i=0 finished running operations in the middle of backfill. retErr: failed to DELETE and INSERT: pq: batch timestamp 1738259733.504503667,0 must be after replica GC threshold 1738259740.400389608,0 (r77: /Table/107{-/2})
    backfill_protected_timestamp_test.go:474: backfill goroutine failed: could not read jobs table: context canceled
    backfill_protected_timestamp_test.go:515: 
        	Error Trace:	pkg/sql/backfill_protected_timestamp_test.go:515
        	Error:      	Received unexpected error:
        	            	failed to DELETE and INSERT: pq: batch timestamp 1738259733.504503667,0 must be after replica GC threshold 1738259740.400389608,0 (r77: /Table/107{-/2})
        	            	(1) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/sql_test.TestBackfillWithProtectedTS.func7.2.1
        	            	  | 	pkg/sql/backfill_protected_timestamp_test.go:438
        	            	  | github.com/cockroachdb/cockroach/pkg/sql_test.TestBackfillWithProtectedTS.func7.2
        	            	  | 	pkg/sql/backfill_protected_timestamp_test.go:463
        	            	  | github.com/cockroachdb/cockroach/pkg/sql_test.TestBackfillWithProtectedTS.func7.Group.GoCtx.4
        	            	  | 	pkg/util/ctxgroup/ctxgroup.go:189
        	            	  | golang.org/x/sync/errgroup.(*Group).Go.func1
        	            	  | 	external/org_golang_x_sync/errgroup/errgroup.go:78
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_amd64.s:1695
        	            	Wraps: (2) failed to DELETE and INSERT
        	            	Wraps: (3) pq: batch timestamp 1738259733.504503667,0 must be after replica GC threshold 1738259740.400389608,0 (r77: /Table/107{-/2})
        	            	Error types: (1) *withstack.withStack (2) *errutil.withPrefix (3) *pq.Error
        	Test:       	TestBackfillWithProtectedTS/create_index
    --- FAIL: TestBackfillWithProtectedTS/create_index (423.10s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=25</code>
 - <code>shard=14</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #140137 sql: TestBackfillWithProtectedTS failed [C-test-failure O-robot T-sql-foundations branch-release-24.2.9-rc release-blocker]
- #140043 sql: TestBackfillWithProtectedTS failed [C-test-failure O-robot P-1 T-sql-foundations branch-release-24.1]
- #139991 sql: TestBackfillWithProtectedTS failed [C-test-failure O-robot P-2 T-sql-foundations branch-release-24.2]
- #139901 sql: TestBackfillWithProtectedTS failed [C-test-failure O-robot P-2 T-sql-foundations branch-release-24.3]
- #139862 sql: TestBackfillWithProtectedTS failed [C-test-failure GA-blocker O-robot P-1 T-sql-foundations branch-release-25.1]
- #139829 sql: TestBackfillWithProtectedTS failed [C-test-failure O-robot P-3 T-sql-foundations branch-release-24.3.4-rc]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestBackfillWithProtectedTS.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

sql.TestBackfillWithProtectedTS [failed](https://tanzanite.cluster.engflow.com/invocations/default/00c0f454-052a-477d-9502-e2d913a983b2?testReportRun=23&testReportShard=14&testReportAttempt=1#targets-Ly9wa2cvc3FsOnNxbF90ZXN0) on master @ [507c68f8abb0c9a3f527c66cb42129c2cb096e7d](https://github.com/cockroachdb/cockroach/commits/507c68f8abb0c9a3f527c66cb42129c2cb096e7d):


```
=== RUN   TestBackfillWithProtectedTS
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestBackfillWithProtectedTS345770608
    test_log_scope.go:76: use -show-logs to present logs inline
    test_server_shim.go:141: cluster virtualization disabled in global scope due to issue: #76378 (expected label: C-bug)
    backfill_protected_timestamp_test.go:526: -- test log scope end --
test logs left over in: outputs.zip/logTestBackfillWithProtectedTS345770608
--- FAIL: TestBackfillWithProtectedTS (494.33s)
=== RUN   TestBackfillWithProtectedTS/create_index
    backfill_protected_timestamp_test.go:477: running backfill with PTS not setup early enough
    backfill_protected_timestamp_test.go:425: i=0 begin running operations in the middle of backfill
    backfill_protected_timestamp_test.go:429: i=0 finished running operations in the middle of backfill. retErr: failed to DELETE and INSERT: pq: batch timestamp 1738563801.384158058,0 must be after replica GC threshold 1738563819.259859579,0 (r78: /Table/107{-/2})
    backfill_protected_timestamp_test.go:474: backfill goroutine failed: could not read jobs table: context canceled
    backfill_protected_timestamp_test.go:515: 
        	Error Trace:	pkg/sql/backfill_protected_timestamp_test.go:515
        	Error:      	Received unexpected error:
        	            	failed to DELETE and INSERT: pq: batch timestamp 1738563801.384158058,0 must be after replica GC threshold 1738563819.259859579,0 (r78: /Table/107{-/2})
        	            	(1) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/sql_test.TestBackfillWithProtectedTS.func7.2.1
        	            	  | 	pkg/sql/backfill_protected_timestamp_test.go:438
        	            	  | github.com/cockroachdb/cockroach/pkg/sql_test.TestBackfillWithProtectedTS.func7.2
        	            	  | 	pkg/sql/backfill_protected_timestamp_test.go:463
        	            	  | github.com/cockroachdb/cockroach/pkg/sql_test.TestBackfillWithProtectedTS.func7.Group.GoCtx.4
        	            	  | 	pkg/util/ctxgroup/ctxgroup.go:189
        	            	  | golang.org/x/sync/errgroup.(*Group).Go.func1
        	            	  | 	external/org_golang_x_sync/errgroup/errgroup.go:78
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_amd64.s:1695
        	            	Wraps: (2) failed to DELETE and INSERT
        	            	Wraps: (3) pq: batch timestamp 1738563801.384158058,0 must be after replica GC threshold 1738563819.259859579,0 (r78: /Table/107{-/2})
        	            	Error types: (1) *withstack.withStack (2) *errutil.withPrefix (3) *pq.Error
        	Test:       	TestBackfillWithProtectedTS/create_index
    --- FAIL: TestBackfillWithProtectedTS/create_index (429.21s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=23</code>
 - <code>shard=14</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #140043 sql: TestBackfillWithProtectedTS failed [C-test-failure O-robot P-1 T-sql-foundations branch-release-24.1]
- #139991 sql: TestBackfillWithProtectedTS failed [C-test-failure O-robot P-2 T-sql-foundations branch-release-24.2]
- #139901 sql: TestBackfillWithProtectedTS failed [C-test-failure O-robot P-2 T-sql-foundations branch-release-24.3]
- #139862 sql: TestBackfillWithProtectedTS failed [C-test-failure GA-blocker O-robot P-1 T-sql-foundations branch-release-25.1]
- #139829 sql: TestBackfillWithProtectedTS failed [C-test-failure O-robot P-3 T-sql-foundations branch-release-24.3.4-rc]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestBackfillWithProtectedTS.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

sql.TestBackfillWithProtectedTS [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/18725905?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/18725905?buildTab=artifacts#/) on master @ [803ecba3e3e48195f256dff9aaea9975c1f0844b](https://github.com/cockroachdb/cockroach/commits/803ecba3e3e48195f256dff9aaea9975c1f0844b):


```
=== RUN   TestBackfillWithProtectedTS
I250204 18:53:48.450735 1 (gostd) rand.go:256  [-] 1  random seed: 9046371300302816388
    test_log_scope.go:165: test logs captured to: /artifacts/tmp/_tmp/ebab9da743cf8d3e9fbccfacc75c4814/logTestBackfillWithProtectedTS2777219903
    test_log_scope.go:76: use -show-logs to present logs inline
    test_server_shim.go:141: cluster virtualization disabled in global scope due to issue: #76378 (expected label: C-bug)
    backfill_protected_timestamp_test.go:526: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/ebab9da743cf8d3e9fbccfacc75c4814/logTestBackfillWithProtectedTS2777219903
--- FAIL: TestBackfillWithProtectedTS (473.50s)
=== RUN   TestBackfillWithProtectedTS/create_index
    backfill_protected_timestamp_test.go:477: running backfill with PTS not setup early enough
    backfill_protected_timestamp_test.go:425: i=0 begin running operations in the middle of backfill
    backfill_protected_timestamp_test.go:429: i=0 finished running operations in the middle of backfill. retErr: failed to DELETE and INSERT: pq: batch timestamp 1738695293.557825841,0 must be after replica GC threshold 1738695302.812017014,0 (r79: /{Table/107/3-Max})
    backfill_protected_timestamp_test.go:474: backfill goroutine failed: expected error was not hit
    backfill_protected_timestamp_test.go:515: 
        	Error Trace:	pkg/sql/backfill_protected_timestamp_test.go:515
        	Error:      	Received unexpected error:
        	            	failed to DELETE and INSERT: pq: batch timestamp 1738695293.557825841,0 must be after replica GC threshold 1738695302.812017014,0 (r79: /{Table/107/3-Max})
        	            	(1) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/sql_test.TestBackfillWithProtectedTS.func7.2.1
        	            	  | 	pkg/sql/backfill_protected_timestamp_test.go:438
        	            	  | github.com/cockroachdb/cockroach/pkg/sql_test.TestBackfillWithProtectedTS.func7.2
        	            	  | 	pkg/sql/backfill_protected_timestamp_test.go:463
        	            	  | github.com/cockroachdb/cockroach/pkg/sql_test.TestBackfillWithProtectedTS.func7.Group.GoCtx.4
        	            	  | 	pkg/util/ctxgroup/ctxgroup.go:189
        	            	  | golang.org/x/sync/errgroup.(*Group).Go.func1
        	            	  | 	external/org_golang_x_sync/errgroup/errgroup.go:78
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_arm64.s:1222
        	            	Wraps: (2) failed to DELETE and INSERT
        	            	Wraps: (3) pq: batch timestamp 1738695293.557825841,0 must be after replica GC threshold 1738695302.812017014,0 (r79: /{Table/107/3-Max})
        	            	Error types: (1) *withstack.withStack (2) *errutil.withPrefix (3) *pq.Error
        	Test:       	TestBackfillWithProtectedTS/create_index
    --- FAIL: TestBackfillWithProtectedTS/create_index (416.53s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #140043 sql: TestBackfillWithProtectedTS failed [C-test-failure O-robot P-1 T-sql-foundations branch-release-24.1]
- #139991 sql: TestBackfillWithProtectedTS failed [C-test-failure O-robot P-2 T-sql-foundations branch-release-24.2]
- #139901 sql: TestBackfillWithProtectedTS failed [C-test-failure O-robot P-2 T-sql-foundations branch-release-24.3]
- #139862 sql: TestBackfillWithProtectedTS failed [C-test-failure O-robot P-1 T-sql-foundations branch-release-25.1]
- #139829 sql: TestBackfillWithProtectedTS failed [C-test-failure O-robot P-3 T-sql-foundations branch-release-24.3.4-rc]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestBackfillWithProtectedTS.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

sql.TestBackfillWithProtectedTS [failed](https://tanzanite.cluster.engflow.com/invocations/default/fcd1434c-2b66-4a39-ae86-da533a25f490?testReportRun=2&testReportShard=14&testReportAttempt=1#targets-Ly9wa2cvc3FsOnNxbF90ZXN0) on master @ [82b1a2b174dcedfd23b0a3a40c6069143f4b038f](https://github.com/cockroachdb/cockroach/commits/82b1a2b174dcedfd23b0a3a40c6069143f4b038f):


```
=== RUN   TestBackfillWithProtectedTS
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestBackfillWithProtectedTS2697217821
    test_log_scope.go:76: use -show-logs to present logs inline
    test_server_shim.go:141: cluster virtualization disabled in global scope due to issue: #76378 (expected label: C-bug)
    backfill_protected_timestamp_test.go:526: -- test log scope end --
test logs left over in: outputs.zip/logTestBackfillWithProtectedTS2697217821
--- FAIL: TestBackfillWithProtectedTS (491.31s)
=== RUN   TestBackfillWithProtectedTS/create_index
    backfill_protected_timestamp_test.go:477: running backfill with PTS not setup early enough
    backfill_protected_timestamp_test.go:425: i=0 begin running operations in the middle of backfill
    backfill_protected_timestamp_test.go:429: i=0 finished running operations in the middle of backfill. retErr: failed to DELETE and INSERT: pq: batch timestamp 1738701097.357997094,0 must be after replica GC threshold 1738701132.289629257,0 (r77: /Table/107{-/2})
    backfill_protected_timestamp_test.go:474: backfill goroutine failed: could not read jobs table: context canceled
    backfill_protected_timestamp_test.go:515: 
        	Error Trace:	pkg/sql/backfill_protected_timestamp_test.go:515
        	Error:      	Received unexpected error:
        	            	failed to DELETE and INSERT: pq: batch timestamp 1738701097.357997094,0 must be after replica GC threshold 1738701132.289629257,0 (r77: /Table/107{-/2})
        	            	(1) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/sql_test.TestBackfillWithProtectedTS.func7.2.1
        	            	  | 	pkg/sql/backfill_protected_timestamp_test.go:438
        	            	  | github.com/cockroachdb/cockroach/pkg/sql_test.TestBackfillWithProtectedTS.func7.2
        	            	  | 	pkg/sql/backfill_protected_timestamp_test.go:463
        	            	  | github.com/cockroachdb/cockroach/pkg/sql_test.TestBackfillWithProtectedTS.func7.Group.GoCtx.4
        	            	  | 	pkg/util/ctxgroup/ctxgroup.go:189
        	            	  | golang.org/x/sync/errgroup.(*Group).Go.func1
        	            	  | 	external/org_golang_x_sync/errgroup/errgroup.go:78
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_amd64.s:1695
        	            	Wraps: (2) failed to DELETE and INSERT
        	            	Wraps: (3) pq: batch timestamp 1738701097.357997094,0 must be after replica GC threshold 1738701132.289629257,0 (r77: /Table/107{-/2})
        	            	Error types: (1) *withstack.withStack (2) *errutil.withPrefix (3) *pq.Error
        	Test:       	TestBackfillWithProtectedTS/create_index
    --- FAIL: TestBackfillWithProtectedTS/create_index (446.31s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=2</code>
 - <code>shard=14</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #140043 sql: TestBackfillWithProtectedTS failed [C-test-failure O-robot P-1 T-sql-foundations branch-release-24.1]
- #139991 sql: TestBackfillWithProtectedTS failed [C-test-failure O-robot P-2 T-sql-foundations branch-release-24.2]
- #139901 sql: TestBackfillWithProtectedTS failed [C-test-failure O-robot P-2 T-sql-foundations branch-release-24.3]
- #139862 sql: TestBackfillWithProtectedTS failed [C-test-failure O-robot P-1 T-sql-foundations branch-release-25.1]
- #139829 sql: TestBackfillWithProtectedTS failed [C-test-failure O-robot P-3 T-sql-foundations branch-release-24.3.4-rc]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestBackfillWithProtectedTS.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

sql.TestBackfillWithProtectedTS [failed](https://tanzanite.cluster.engflow.com/invocations/default/8a7647cd-8d67-4f6b-828f-4755c65f4db4?testReportRun=17&testReportShard=14&testReportAttempt=1#targets-Ly9wa2cvc3FsOnNxbF90ZXN0) on master @ [d276833ff08ef11d7894b7b0124f1883d90a3377](https://github.com/cockroachdb/cockroach/commits/d276833ff08ef11d7894b7b0124f1883d90a3377):


```
=== RUN   TestBackfillWithProtectedTS
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestBackfillWithProtectedTS3881173975
    test_log_scope.go:76: use -show-logs to present logs inline
    test_server_shim.go:141: cluster virtualization disabled in global scope due to issue: #76378 (expected label: C-bug)
    backfill_protected_timestamp_test.go:526: -- test log scope end --
test logs left over in: outputs.zip/logTestBackfillWithProtectedTS3881173975
--- FAIL: TestBackfillWithProtectedTS (479.63s)
=== RUN   TestBackfillWithProtectedTS/create_index
    backfill_protected_timestamp_test.go:477: running backfill with PTS not setup early enough
    backfill_protected_timestamp_test.go:425: i=0 begin running operations in the middle of backfill
    backfill_protected_timestamp_test.go:429: i=0 finished running operations in the middle of backfill. retErr: failed to DELETE and INSERT: pq: batch timestamp 1738736794.079181834,0 must be after replica GC threshold 1738736795.878992897,0 (r81: /Table/107/{1/1044205956791959553-2})
    backfill_protected_timestamp_test.go:474: backfill goroutine failed: could not read jobs table: context canceled
    backfill_protected_timestamp_test.go:515: 
        	Error Trace:	pkg/sql/backfill_protected_timestamp_test.go:515
        	Error:      	Received unexpected error:
        	            	failed to DELETE and INSERT: pq: batch timestamp 1738736794.079181834,0 must be after replica GC threshold 1738736795.878992897,0 (r81: /Table/107/{1/1044205956791959553-2})
        	            	(1) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/sql_test.TestBackfillWithProtectedTS.func7.2.1
        	            	  | 	pkg/sql/backfill_protected_timestamp_test.go:438
        	            	  | github.com/cockroachdb/cockroach/pkg/sql_test.TestBackfillWithProtectedTS.func7.2
        	            	  | 	pkg/sql/backfill_protected_timestamp_test.go:463
        	            	  | github.com/cockroachdb/cockroach/pkg/sql_test.TestBackfillWithProtectedTS.func7.Group.GoCtx.4
        	            	  | 	pkg/util/ctxgroup/ctxgroup.go:189
        	            	  | golang.org/x/sync/errgroup.(*Group).Go.func1
        	            	  | 	external/org_golang_x_sync/errgroup/errgroup.go:78
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_amd64.s:1695
        	            	Wraps: (2) failed to DELETE and INSERT
        	            	Wraps: (3) pq: batch timestamp 1738736794.079181834,0 must be after replica GC threshold 1738736795.878992897,0 (r81: /Table/107/{1/1044205956791959553-2})
        	            	Error types: (1) *withstack.withStack (2) *errutil.withPrefix (3) *pq.Error
        	Test:       	TestBackfillWithProtectedTS/create_index
    --- FAIL: TestBackfillWithProtectedTS/create_index (431.11s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=17</code>
 - <code>shard=14</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #140043 sql: TestBackfillWithProtectedTS failed [C-test-failure O-robot P-1 T-sql-foundations branch-release-24.1]
- #139991 sql: TestBackfillWithProtectedTS failed [C-test-failure O-robot P-2 T-sql-foundations branch-release-24.2]
- #139901 sql: TestBackfillWithProtectedTS failed [C-test-failure O-robot P-2 T-sql-foundations branch-release-24.3]
- #139862 sql: TestBackfillWithProtectedTS failed [C-test-failure O-robot P-1 T-sql-foundations branch-release-25.1]
- #139829 sql: TestBackfillWithProtectedTS failed [C-test-failure O-robot P-3 T-sql-foundations branch-release-24.3.4-rc]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestBackfillWithProtectedTS.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

sql.TestBackfillWithProtectedTS [failed](https://tanzanite.cluster.engflow.com/invocations/default/f996ea8e-bd64-45d4-8748-8adb5264d0a9?testReportRun=9&testReportShard=14&testReportAttempt=1#targets-Ly9wa2cvc3FsOnNxbF90ZXN0) on master @ [13d8c5191614530b50802f77081f15d2ce2ca705](https://github.com/cockroachdb/cockroach/commits/13d8c5191614530b50802f77081f15d2ce2ca705):


```
=== RUN   TestBackfillWithProtectedTS
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestBackfillWithProtectedTS1511645903
    test_log_scope.go:76: use -show-logs to present logs inline
    test_server_shim.go:141: cluster virtualization disabled in global scope due to issue: #76378 (expected label: C-bug)
    backfill_protected_timestamp_test.go:526: -- test log scope end --
test logs left over in: outputs.zip/logTestBackfillWithProtectedTS1511645903
--- FAIL: TestBackfillWithProtectedTS (492.20s)
=== RUN   TestBackfillWithProtectedTS/create_index
    backfill_protected_timestamp_test.go:477: running backfill with PTS not setup early enough
    backfill_protected_timestamp_test.go:425: i=0 begin running operations in the middle of backfill
    backfill_protected_timestamp_test.go:429: i=0 finished running operations in the middle of backfill. retErr: failed to DELETE and INSERT: pq: batch timestamp 1738960285.200871388,0 must be after replica GC threshold 1738960313.842443276,0 (r79: /{Table/107/3-Max})
    backfill_protected_timestamp_test.go:474: backfill goroutine failed: expected error was not hit
    backfill_protected_timestamp_test.go:515: 
        	Error Trace:	pkg/sql/backfill_protected_timestamp_test.go:515
        	Error:      	Received unexpected error:
        	            	failed to DELETE and INSERT: pq: batch timestamp 1738960285.200871388,0 must be after replica GC threshold 1738960313.842443276,0 (r79: /{Table/107/3-Max})
        	            	(1) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/sql_test.TestBackfillWithProtectedTS.func7.2.1
        	            	  | 	pkg/sql/backfill_protected_timestamp_test.go:438
        	            	  | github.com/cockroachdb/cockroach/pkg/sql_test.TestBackfillWithProtectedTS.func7.2
        	            	  | 	pkg/sql/backfill_protected_timestamp_test.go:463
        	            	  | github.com/cockroachdb/cockroach/pkg/sql_test.TestBackfillWithProtectedTS.func7.Group.GoCtx.4
        	            	  | 	pkg/util/ctxgroup/ctxgroup.go:189
        	            	  | golang.org/x/sync/errgroup.(*Group).Go.func1
        	            	  | 	external/org_golang_x_sync/errgroup/errgroup.go:78
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_amd64.s:1695
        	            	Wraps: (2) failed to DELETE and INSERT
        	            	Wraps: (3) pq: batch timestamp 1738960285.200871388,0 must be after replica GC threshold 1738960313.842443276,0 (r79: /{Table/107/3-Max})
        	            	Error types: (1) *withstack.withStack (2) *errutil.withPrefix (3) *pq.Error
        	Test:       	TestBackfillWithProtectedTS/create_index
    --- FAIL: TestBackfillWithProtectedTS/create_index (450.69s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=9</code>
 - <code>shard=14</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #140708 sql: TestBackfillWithProtectedTS failed [C-test-failure O-robot T-sql-foundations branch-release-25.1.0-rc release-blocker]
- #140043 sql: TestBackfillWithProtectedTS failed [C-test-failure O-robot P-1 T-sql-foundations branch-release-24.1]
- #139991 sql: TestBackfillWithProtectedTS failed [C-test-failure O-robot P-2 T-sql-foundations branch-release-24.2]
- #139901 sql: TestBackfillWithProtectedTS failed [C-test-failure O-robot P-2 T-sql-foundations branch-release-24.3]
- #139862 sql: TestBackfillWithProtectedTS failed [C-test-failure O-robot P-1 T-sql-foundations branch-release-25.1]
- #139829 sql: TestBackfillWithProtectedTS failed [C-test-failure O-robot P-3 T-sql-foundations branch-release-24.3.4-rc]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestBackfillWithProtectedTS.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

sql.TestBackfillWithProtectedTS [failed](https://tanzanite.cluster.engflow.com/invocations/default/d3ccecba-14a9-4b61-aca2-d207b1d71b94?testReportRun=10&testReportShard=14&testReportAttempt=1#targets-Ly9wa2cvc3FsOnNxbF90ZXN0) on master @ [ecfb74b7ad75327f11814966a2cdab1b9793a549](https://github.com/cockroachdb/cockroach/commits/ecfb74b7ad75327f11814966a2cdab1b9793a549):


```
=== RUN   TestBackfillWithProtectedTS
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestBackfillWithProtectedTS2521234472
    test_log_scope.go:76: use -show-logs to present logs inline
    test_server_shim.go:141: cluster virtualization disabled in global scope due to issue: #76378 (expected label: C-bug)
    backfill_protected_timestamp_test.go:526: -- test log scope end --
test logs left over in: outputs.zip/logTestBackfillWithProtectedTS2521234472
--- FAIL: TestBackfillWithProtectedTS (486.61s)
=== RUN   TestBackfillWithProtectedTS/create_index
    backfill_protected_timestamp_test.go:477: running backfill with PTS not setup early enough
    backfill_protected_timestamp_test.go:425: i=0 begin running operations in the middle of backfill
    backfill_protected_timestamp_test.go:429: i=0 finished running operations in the middle of backfill. retErr: failed to DELETE and INSERT: pq: batch timestamp 1739081999.539580442,0 must be after replica GC threshold 1739082014.349261007,0 (r77: /Table/107{-/2})
    backfill_protected_timestamp_test.go:474: backfill goroutine failed: could not read jobs table: context canceled
    backfill_protected_timestamp_test.go:515: 
        	Error Trace:	pkg/sql/backfill_protected_timestamp_test.go:515
        	Error:      	Received unexpected error:
        	            	failed to DELETE and INSERT: pq: batch timestamp 1739081999.539580442,0 must be after replica GC threshold 1739082014.349261007,0 (r77: /Table/107{-/2})
        	            	(1) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/sql_test.TestBackfillWithProtectedTS.func7.2.1
        	            	  | 	pkg/sql/backfill_protected_timestamp_test.go:438
        	            	  | github.com/cockroachdb/cockroach/pkg/sql_test.TestBackfillWithProtectedTS.func7.2
        	            	  | 	pkg/sql/backfill_protected_timestamp_test.go:463
        	            	  | github.com/cockroachdb/cockroach/pkg/sql_test.TestBackfillWithProtectedTS.func7.Group.GoCtx.4
        	            	  | 	pkg/util/ctxgroup/ctxgroup.go:189
        	            	  | golang.org/x/sync/errgroup.(*Group).Go.func1
        	            	  | 	external/org_golang_x_sync/errgroup/errgroup.go:78
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_amd64.s:1695
        	            	Wraps: (2) failed to DELETE and INSERT
        	            	Wraps: (3) pq: batch timestamp 1739081999.539580442,0 must be after replica GC threshold 1739082014.349261007,0 (r77: /Table/107{-/2})
        	            	Error types: (1) *withstack.withStack (2) *errutil.withPrefix (3) *pq.Error
        	Test:       	TestBackfillWithProtectedTS/create_index
    --- FAIL: TestBackfillWithProtectedTS/create_index (422.23s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=10</code>
 - <code>shard=14</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #140780 sql: TestBackfillWithProtectedTS failed [C-test-failure O-robot P-1 T-sql-foundations branch-release-24.3.5-rc]
- #140708 sql: TestBackfillWithProtectedTS failed [C-test-failure O-robot T-sql-foundations branch-release-25.1.0-rc release-blocker]
- #140043 sql: TestBackfillWithProtectedTS failed [C-test-failure O-robot P-1 T-sql-foundations branch-release-24.1]
- #139991 sql: TestBackfillWithProtectedTS failed [C-test-failure O-robot P-2 T-sql-foundations branch-release-24.2]
- #139901 sql: TestBackfillWithProtectedTS failed [C-test-failure O-robot P-2 T-sql-foundations branch-release-24.3]
- #139862 sql: TestBackfillWithProtectedTS failed [C-test-failure O-robot P-1 T-sql-foundations branch-release-25.1]
- #139829 sql: TestBackfillWithProtectedTS failed [C-test-failure O-robot P-3 T-sql-foundations branch-release-24.3.4-rc]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestBackfillWithProtectedTS.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

sql.TestBackfillWithProtectedTS [failed](https://tanzanite.cluster.engflow.com/invocations/default/cc2cfc3c-43bb-4caa-8ea9-071448ad1683?testReportRun=12&testReportShard=14&testReportAttempt=1#targets-Ly9wa2cvc3FsOnNxbF90ZXN0) on master @ [5e499490a29405b71d5b2d568a43fb7e6b14fe56](https://github.com/cockroachdb/cockroach/commits/5e499490a29405b71d5b2d568a43fb7e6b14fe56):


```
=== RUN   TestBackfillWithProtectedTS
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestBackfillWithProtectedTS7052065
    test_log_scope.go:76: use -show-logs to present logs inline
    test_server_shim.go:141: cluster virtualization disabled in global scope due to issue: #76378 (expected label: C-bug)
    backfill_protected_timestamp_test.go:526: -- test log scope end --
test logs left over in: outputs.zip/logTestBackfillWithProtectedTS7052065
--- FAIL: TestBackfillWithProtectedTS (509.94s)
=== RUN   TestBackfillWithProtectedTS/create_index
    backfill_protected_timestamp_test.go:477: running backfill with PTS not setup early enough
    backfill_protected_timestamp_test.go:425: i=0 begin running operations in the middle of backfill
    backfill_protected_timestamp_test.go:429: i=0 finished running operations in the middle of backfill. retErr: failed to DELETE and INSERT: pq: batch timestamp 1739220697.377542421,0 must be after replica GC threshold 1739220735.700163390,0 (r79: /{Table/107/3-Max})
    backfill_protected_timestamp_test.go:474: backfill goroutine failed: expected error was not hit
    backfill_protected_timestamp_test.go:515: 
        	Error Trace:	pkg/sql/backfill_protected_timestamp_test.go:515
        	Error:      	Received unexpected error:
        	            	failed to DELETE and INSERT: pq: batch timestamp 1739220697.377542421,0 must be after replica GC threshold 1739220735.700163390,0 (r79: /{Table/107/3-Max})
        	            	(1) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/sql_test.TestBackfillWithProtectedTS.func7.2.1
        	            	  | 	pkg/sql/backfill_protected_timestamp_test.go:438
        	            	  | github.com/cockroachdb/cockroach/pkg/sql_test.TestBackfillWithProtectedTS.func7.2
        	            	  | 	pkg/sql/backfill_protected_timestamp_test.go:463
        	            	  | github.com/cockroachdb/cockroach/pkg/sql_test.TestBackfillWithProtectedTS.func7.Group.GoCtx.4
        	            	  | 	pkg/util/ctxgroup/ctxgroup.go:189
        	            	  | golang.org/x/sync/errgroup.(*Group).Go.func1
        	            	  | 	external/org_golang_x_sync/errgroup/errgroup.go:78
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_amd64.s:1695
        	            	Wraps: (2) failed to DELETE and INSERT
        	            	Wraps: (3) pq: batch timestamp 1739220697.377542421,0 must be after replica GC threshold 1739220735.700163390,0 (r79: /{Table/107/3-Max})
        	            	Error types: (1) *withstack.withStack (2) *errutil.withPrefix (3) *pq.Error
        	Test:       	TestBackfillWithProtectedTS/create_index
    --- FAIL: TestBackfillWithProtectedTS/create_index (465.33s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=12</code>
 - <code>shard=14</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #140780 sql: TestBackfillWithProtectedTS failed [C-test-failure O-robot P-1 T-sql-foundations branch-release-24.3.5-rc]
- #140708 sql: TestBackfillWithProtectedTS failed [C-test-failure O-robot P-2 T-sql-foundations branch-release-25.1.0-rc]
- #140043 sql: TestBackfillWithProtectedTS failed [C-test-failure O-robot P-1 T-sql-foundations branch-release-24.1]
- #139991 sql: TestBackfillWithProtectedTS failed [C-test-failure O-robot P-2 T-sql-foundations branch-release-24.2]
- #139901 sql: TestBackfillWithProtectedTS failed [C-test-failure O-robot P-2 T-sql-foundations branch-release-24.3]
- #139862 sql: TestBackfillWithProtectedTS failed [C-test-failure O-robot P-1 T-sql-foundations branch-release-25.1]
- #139829 sql: TestBackfillWithProtectedTS failed [C-test-failure O-robot P-3 T-sql-foundations branch-release-24.3.4-rc]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestBackfillWithProtectedTS.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/pkg/sql/backfill_protected_timestamp_test.go
+++ b/pkg/sql/backfill_protected_timestamp_test.go
@@ -233,13 +233,13 @@ func TestBackfillWithProtectedTS(t *testing.T) {
 	skip.UnderDeadlock(t, "test takes too long")
 	skip.UnderStress(t, "test takes too long")
 	skip.UnderRace(t, "test takes too long")
-	skip.WithIssue(t, 139862)
 
 	ctx := context.Background()
 	backfillQueryWait := make(chan struct{})
 	backfillQueryResume := make(chan struct{})
 	blockBackFillsForPTSFailure := atomic.Bool{}
 	blockBackFillsForPTSCheck := atomic.Bool{}
+	usingDeclarativeSchemaChanger := atomic.Bool{}
 	var s serverutils.TestServerInterface
 	var db *gosql.DB
 	var tableID uint32
@@ -253,7 +253,7 @@ func TestBackfillWithProtectedTS(t *testing.T) {
 					// Cause the backfill to pause before adding the protected
 					// timestamp. This knob is for testing schema changes that
 					// are on the declarative schema changer.
-					if blockBackFillsForPTSFailure.Load() {
+					if blockBackFillsForPTSFailure.Load() && usingDeclarativeSchemaChanger.Load() {
 						if !blockBackFillsForPTSFailure.Swap(false) {
 							return nil
 						}
@@ -268,7 +268,7 @@ func TestBackfillWithProtectedTS(t *testing.T) {
 					// Cause the backfill to pause after it already began running
 					// and has installed a protected timestamp. This knob is for
 					// testing schema changes that use the index backfiller.
-					if blockBackFillsForPTSCheck.Load() {
+					if blockBackFillsForPTSCheck.Load() && usingDeclarativeSchemaChanger.Load() {
 						_, prefix, err := s.Codec().DecodeTablePrefix(sp.Key)
 						if err != nil || prefix != tableID {
 							//nolint:returnerrcheck
@@ -287,7 +287,7 @@ func TestBackfillWithProtectedTS(t *testing.T) {
 				RunBeforeQueryBackfill: func() error {
 					// Cause the backfill to pause before adding the protected
 					// timestamp. This knob is for testing CREATE MATERIALIZED VIEW.
-					if blockBackFillsForPTSFailure.Load() {
+					if blockBackFillsForPTSFailure.Load() && !usingDeclarativeSchemaChanger.Load() {
 						if !blockBackFillsForPTSFailure.Swap(false) {
 							return nil
 						}
@@ -302,7 +302,7 @@ func TestBackfillWithProtectedTS(t *testing.T) {
 					// Detect the first scan on table from the backfill, which is
 					// after the PTS has been set up. This knob is for testing CREATE
 					// MATERIALIZED VIEW.
-					if blockBackFillsForPTSCheck.Load() &&
+					if blockBackFillsForPTSCheck.Load() && !usingDeclarativeSchemaChanger.Load() &&
 						request.Txn != nil &&
 						request.Txn.Name == "schemaChangerBackfill" &&
 						request.Requests[0].GetInner().Method() == kvpb.Scan {
@@ -368,47 +368,57 @@ func TestBackfillWithProtectedTS(t *testing.T) {
 	} {
 		rSys.Exec(t, sql)
 	}
+	for _, sql := range []string{
+		"SET CLUSTER SETTING sql.stats.automatic_collection.enabled = false",
+		"ALTER DATABASE defaultdb CONFIGURE ZONE USING gc.ttlseconds = 5",
+	} {
+		r.Exec(t, sql)
+	}
 
 	const initialRowCount = 500000
 	const rowsDeletedPerIteration = 200000
 	const rowsAddedPerIteration = 1
 
 	for _, tc := range []struct {
-		name                 string
-		backfillSchemaChange string
-		jobDescriptionPrefix string
-		postTestQuery        string
-		expectedCount        int
+		name                   string
+		tableName              string
+		backfillSchemaChange   string
+		jobDescriptionPrefix   string
+		postTestQuery          string
+		expectedCount          int
+		usingDeclSchemaChanger bool
 	}{
 		{
-			name:                 "create materialized view",
-			backfillSchemaChange: "CREATE MATERIALIZED VIEW test AS (SELECT n from t)",
-			jobDescriptionPrefix: "CREATE MATERIALIZED VIEW",
-			postTestQuery:        "SELECT count(*) FROM test",
-			expectedCount:        initialRowCount - rowsDeletedPerIteration + rowsAddedPerIteration,
+			name:                   "create materialized view",
+			tableName:              "t_mat_view",
+			backfillSchemaChange:   "CREATE MATERIALIZED VIEW test AS (SELECT n from t_mat_view)",
+			jobDescriptionPrefix:   "CREATE MATERIALIZED VIEW",
+			postTestQuery:          "SELECT count(*) FROM test",
+			expectedCount:          initialRowCount - rowsDeletedPerIteration + rowsAddedPerIteration,
+			usingDeclSchemaChanger: false,
 		},
 		{
-			name:                 "create index",
-			backfillSchemaChange: "CREATE INDEX idx ON t(n)",
-			jobDescriptionPrefix: "CREATE INDEX idx",
-			postTestQuery:        "SELECT count(*) FROM t@idx",
-			expectedCount:        initialRowCount - 2*rowsDeletedPerIteration + 2*rowsAddedPerIteration,
+			name:                   "create index",
+			tableName:              "t_idx",
+			backfillSchemaChange:   "CREATE INDEX idx ON t_idx(n)",
+			jobDescriptionPrefix:   "CREATE INDEX idx",
+			postTestQuery:          "SELECT count(*) FROM t_idx@idx",
+			expectedCount:          initialRowCount - 2*rowsDeletedPerIteration + 2*rowsAddedPerIteration,
+			usingDeclSchemaChanger: true,
 		},
 	} {
 		t.Run(tc.name, func(t *testing.T) {
+			usingDeclarativeSchemaChanger.Store(tc.usingDeclSchemaChanger)
 			for _, sql := range []string{
-				"SET CLUSTER SETTING sql.stats.automatic_collection.enabled = false",
-				"ALTER DATABASE defaultdb CONFIGURE ZONE USING gc.ttlseconds = 5",
-				"DROP TABLE IF EXISTS t CASCADE",
-				"CREATE TABLE t(n int)",
-				"ALTER TABLE t CONFIGURE ZONE USING range_min_bytes = 0, range_max_bytes = 67108864, gc.ttlseconds = 5",
-				fmt.Sprintf("INSERT INTO t(n) SELECT * FROM generate_series(1, %d)", initialRowCount),
+				fmt.Sprintf("CREATE TABLE %s(n int primary key)", tc.tableName),
+				fmt.Sprintf("ALTER TABLE %s CONFIGURE ZONE USING range_min_bytes = 0, range_max_bytes = 67108864, gc.ttlseconds = 5", tc.tableName),
+				fmt.Sprintf("INSERT INTO %s(n) SELECT * FROM generate_series(1, %d)", tc.tableName, initialRowCount),
 			} {
 				r.Exec(t, sql)
 			}
 
 			getTableID := func() (tableID uint32) {
-				r.QueryRow(t, `SELECT 't'::regclass::oid`).Scan(&tableID)
+				r.QueryRow(t, fmt.Sprintf(`SELECT '%s'::regclass::oid`, tc.tableName)).Scan(&tableID)
 				return tableID
 			}
 			tableID = getTableID()
@@ -423,20 +433,36 @@ func TestBackfillWithProtectedTS(t *testing.T) {
 				for i := 0; i < 2; i++ {
 					i := i
 					if err := func() (retErr error) {
-						t.Logf("i=%d begin running operations in the middle of backfill", i)
 						<-backfillQueryWait
+						t.Logf(
+							"%s i=%d begin running operations in the middle of backfill",
+							timeutil.Now().Format(time.RFC3339), i,
+						)
 						defer func() {
+							t.Logf(
+								"%s i=%d finished running operations in the middle of backfill. retErr: %v",
+								timeutil.Now().Format(time.RFC3339), i, retErr)
 							backfillQueryResume <- struct{}{}
-							t.Logf("i=%d finished running operations in the middle of backfill. retErr: %v", i, retErr)
 						}()
 						if _, err := db.ExecContext(ctx, "SET sql_safe_updates=off"); err != nil {
 							return errors.Wrap(err, "failed to set sql_safe_updates")
 						}
+						deletedSoFar := 0
+						for deletedSoFar < rowsDeletedPerIteration {
+							const deleteBatchSize = 1000
+							if _, err := db.ExecContext(ctx, fmt.Sprintf(
+								"DELETE FROM %s LIMIT %d",
+								tc.tableName, deleteBatchSize,
+							)); err != nil {
+								return errors.Wrapf(err, "failed to DELETE; deleted %d so far", deletedSoFar)
+							}
+							deletedSoFar += deleteBatchSize
+						}
 						if _, err := db.ExecContext(ctx, fmt.Sprintf(
-							"BEGIN; DELETE FROM t LIMIT %d; INSERT INTO t VALUES('9999999'); COMMIT",
-							rowsDeletedPerIteration,
+							"INSERT INTO %s VALUES(9999999 + %d)",
+							tc.tableName, i,
 						)); err != nil {
-							return errors.Wrap(err, "failed to DELETE and INSERT")
+							return errors.Wrap(err, "failed to INSERT")
 						}
 						if err := refreshTo(ctx, tableKey, ts.Clock().Now()); err != nil {
 							return errors.Wrap(err, "failed to refresh in-memory PTS")
@@ -444,9 +470,9 @@ func TestBackfillWithProtectedTS(t *testing.T) {
 						if err := refreshPTSCacheTo(ctx, ts.Clock().Now()); err != nil {
 							return errors.Wrap(err, "failed to refresh PTS cache")
 						}
-						if _, err := db.ExecContext(ctx, `
+						if _, err := db.ExecContext(ctx, fmt.Sprintf(`
 SELECT crdb_internal.kv_enqueue_replica(range_id, 'mvccGC', true)
-FROM (SELECT range_id FROM [SHOW RANGES FROM TABLE t] ORDER BY start_key);`); err != nil {
+FROM (SELECT range_id FROM [SHOW RANGES FROM TABLE %s] ORDER BY start_key);`, tc.tableName)); err != nil {
 							return errors.Wrap(err, "failed to enqueue replica for GC")
 						}
 						row := db.QueryRowContext(ctx, "SELECT count(*) FROM system.protected_ts_records WHERE meta_type='jobs'")
@@ -471,11 +497,9 @@ FROM (SELECT range_id FROM [SHOW RANGES FROM TABLE t] ORDER BY start_key);`); er
 				// Backfill with the PTS being not setup early enough, which will
 				// lead to failure.
 				defer func() {
-					if retErr != nil {
-						t.Logf("backfill goroutine failed: %v", retErr)
-					}
+					t.Logf("%s backfill goroutine finished. retErr: %v", timeutil.Now().Format(time.RFC3339), retErr)
 				}()
-				t.Logf("running backfill with PTS not setup early enough")
+				t.Logf("%s running backfill with PTS not setup early enough", timeutil.Now().Format(time.RFC3339))
 				blockBackFillsForPTSFailure.Swap(true)
 				_, err := db.ExecContext(ctx, tc.backfillSchemaChange)
 				if err == nil || !testutils.IsError(err, "unable to retry backfill since fixed timestamp is before the GC timestamp") {
@@ -504,7 +528,7 @@ FROM (SELECT range_id FROM [SHOW RANGES FROM TABLE t] ORDER BY start_key);`); er
 				}
 				// Next backfill with the PTS being setup on time, which should always
 				// succeed.
-				t.Logf("running backfill with PTS setup on time")
+				t.Logf("%s running backfill with PTS setup on time", timeutil.Now().Format(time.RFC3339))
 				blockBackFillsForPTSCheck.Swap(true)
 				_, err = db.ExecContext(ctx, tc.backfillSchemaChange)
 				if err != nil {

